### PR TITLE
Allow flexible Prometheus queries in the frontend

### PR DESCRIPTION
**Pull request by miabrahams**
This allows us to write custom queries and send them to Prometheus. It enables us to add multiple charts and run custom queries.

We need to get Prometheus collecting Metrics Server data. Recommended to use kube-prometheus-stack.

https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack

If we want to add a lighter setup that doesn't include Grafana, we could try Kube-Prometheus.

https://github.com/prometheus-operator/kube-prometheus

### CODE REVIEW

File: server/Models/prometheusModel.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const start = new Date();
+  start.setMinutes(end.getMinutes()-range)
+  //range query string to append to base prom fetch
+  return `start=${start.toISOString()}&end=${end.toISOString()}&step=${stepString}`;
+}
+
+
+// {metrics: '' , values[[time,counter],[]....]}
+// fetch from Prometheus
+export async function runPromQuery(query: string, dateString: string) {
```
Comment (MXW9699) [inline server/Models/prometheusModel.ts:30]: add comments about parameters and what each are used for ie: 
@params query - base prometheus query expression to be evaluated.
@params dateString -  time range for the query expression to be evaluated. Following the format of the result of buildDateString()

File: server/Controllers/promMetrics.ts [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
     return next({
-      log: `error in promQuery: ${err}`,
+      log: `error in promQuery. Error: ${err}`,
       status: 400,
       message: { err: 'could not get prom data' },
     });
   }
 };
+
+
```
Comment (MXW9699) [inline server/Controllers/promMetrics.ts]: small comment about what this middleware does, expected output. remove excess console.logs

File: server/Models/k8sModel.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+//creates a Kubernetes cluster object
+import { KubeConfig, CoreV1Api, Metrics, topNodes, topPods, NodeMetric, V1Pod, NodeStatus, V1PodStatus } from '@kubernetes/client-node';
```
Comment (MXW9699) [inline server/Models/k8sModel.ts]: do all of these get used here?

File: server/Controllers/kubeMetrics.ts
Comment (MXW9699) [inline server/Controllers/kubeMetrics.ts:1]: can we import some stuff from k8sModel.ts that has already been defined?

**Review by MXW9699 — COMMENTED** [review]
just general commenting and console.log clean up.

File: server/Models/prometheusModel.ts (hunk 1 above)
Comment (miabrahams) [reply server/Models/prometheusModel.ts:30]: Addressed this in [f009a95](https://github.com/oslabs-beta/KubeSentry/pull/19/commits/f009a95621f086dcebd2afc6ca7a501de94fdf2f)

**Review by MXW9699 — APPROVED** [review]
looks good!

**Pull request merged**